cmake/python.cmake

```diff
@@ -15,6 +15,28 @@ if(WIN32)
   file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
 endif()
 
+# a precomputed PYTHON_INSTALL_DIR can be passed along with the matching PYTHON_EXECUTABLE here
+# in order to avoid needing to call PYTHON_EXECUTABLE in the next block in each package
+# the precomputed value is only taken if the PYTHON_EXECUTABLE matches what was found above
+if(DEFINED PYTHON_INSTALL_DIR_TUPLE)
+  string(REPLACE ":" ";" _tuple ${PYTHON_INSTALL_DIR_TUPLE})
+  list(LENGTH _tuple _tuple_len)
+  if(_tuple_len EQUAL 2)
+    list(GET _tuple 0 _tuple_PYTHON_EXECUTABLE)
+    list(GET _tuple 1 _tuple_PYTHON_INSTALL_DIR)
+    if("${_tuple_PYTHON_EXECUTABLE}" STREQUAL "${PYTHON_EXECUTABLE}")
+      set(PYTHON_INSTALL_DIR "${_tuple_PYTHON_INSTALL_DIR}")
```

Comment:
I thought I was avoiding a conflict where the user could not set `PYTHON_INSTALL_DIR` with `-D` after using `PYTHON_INSTALL_DIR_TUPLE` because the cached version of `PYTHON_INSTALL_DIR_TUPLE` would always cause `PYTHON_INSTALL_DIR` to be set according to its contents. I also expects this to be passed by the build tool, which isn't annoyed by having to send it each time.

I don't think it is necessary to cache this variable, as the `PYTHON_INSTALL_DIR` is cached, but I can do so if you wish.